### ISSUE TITLE
[gtk-3.14] Weird white lines on focused window's buttons/scrollbars one all themes sans *Submarine

### ISSUE THREAD

**Issue by msourada** [mate-desktop/mate-themes#187]
#### Expected behaviour
No rendering artifacts

#### Actual behaviour
Buttons, Tool buttons, GtkScale slider button and GtkScrollbar buttons have mostly horizontal white lines that should not be there. These lines show only when the window is focused. See the attachments.

#### Steps to reproduce the behaviour
Select BlackMate, BlueMenta, Menta, TraditionalGreen or TraditionalOK theme and open gtk3 app that has any of the aforementioned widgets, e.g. awf.

#### MATE general version
N/A – gtk3-3.14.13-20.el7_3.1.x86_64 on XFCE on CentOS 7

#### Package version
git clone of gtk-3.14, commit 8e9d27a

#### Linux Distribution
CentOS 7

#### Additional info
See the attached screenshots:

- blue-menta-focused.png: theme BlueMenta, focused window, issue present
- blue-menta-unfocused.png: theme BlueMenta, unfocused window, everything OK
- blue-submarine.png: theme Blue-Submarine, everything OK

![blue-menta-focused](https://user-images.githubusercontent.com/20648954/27500497-ff109286-5868-11e7-8e70-13a61337fbcd.png)
![blue-menta-unfocused](https://user-images.githubusercontent.com/20648954/27500496-ff0b2526-5868-11e7-9915-e25aac4bf302.png)
![blue-submarine](https://user-images.githubusercontent.com/20648954/27500498-ff11d63c-5868-11e7-888d-1d9b154dded3.png)

**Comment by lukefromdc** [comment]
That looks like an issue with outlines, which are used to permit keyboard navigation. I can no longer build or run MATE with GTK 3.14 on my system as too many things have changed (GTK 3.16 oldest possible rollback), so I can't test this but I can try to point you to the issue. 

Fire up GtkInspector, and with it running watch the offending widgets while you bring them into focus. Not sure where it will show all the style properties in GTK 3.14 (it changed with 3.20), but you are looking for something that changes when focussed. Focus and unfocus while watching the column that lists the style properties and where in the theme they come from. Probably an outline, but could be anything that changes with focus.

**Comment by raveit65** [comment]
I don't see that with gtk-widget-factory on Centos7 MATE session.
What application is shown in your screenshot?
> has any of the aforementioned widgets, e.g. awf.

What is awf ?

**Comment by msourada** [comment]
@raveit65 
AWF is [A Widget Factory](https://github.com/valr/awf), I cannot seem to find The Widget Factory for gtk3, but I see it in other desktop apps (e.g. nautilus, or any xfce that is built agains gtk3, but as far as I can remember, I don't have any ATM on CentOS and Fedora does not have gtk-3.14...) as well.

@lukefromdc 
I'll try to look into it when I have some spare time, thanks for guiding me.

**Comment by raveit65** [comment]
> AWF is A Widget Factory, I cannot seem to find The Widget Factory for gtk3

Install gtk3-devel and at commandline do 'gtk3-widget-factory'
If i have a application to reproduce the issue i can try to fix it. 
I don't use xfce, so 'any xfce that is built agains gtk3' give me no hint :-)
Well i will check nautilus.
Hopefully it isn't specific to using xfce WM

**Comment by raveit65** [comment]
This isn't a outline for using the tab key for keyboard navigation as this is disabled in gtk+-3.14 branch ;-)
https://github.com/mate-desktop/mate-themes/blob/gtk3.16/desktop-themes/BlueMenta/gtk-3.0/gtk-widgets.css#L36
```
outline-width: 0px; /* disable ugly focus-line */
```
And i am sorry to say, i can't reproduce it with latest mate-themes version for 3.14 in rhel7/centos7.
I installed xfce and AWF and i don't see this white lines with using 'awf-gtk3' if the window is focused.
```
$ rpm -qa mate-themes
mate-themes-3.14.6-1.el7.centos.noarch
```
Why you don't use mate-themes from epel7 repo?

**Comment by msourada** [comment]
@raveit65 
sorry for the late reply, I was offline the last two days.

I have the rpm installed and it suffers the same issue – hence I tried if latest upstream is same or not. I wonder, as you cannot reproduce, if it's some wierd misconfiguration on my side or some weird graphics driver bug... I'll try d/l and run CentOS 7 Live Gnome to get as close to the default config as possible and report back.

What prompted me to finally report the bug is the fact that I do not see it in the Sumbarine themes that are from the same theme pack, so I guessed there was some hope in solving it...
